From 84aa51f4d9011014a92cd0299a1f14d253dba9b1 Mon Sep 17 00:00:00 2001
From: ricolin <rlin@vexxhost.com>
Date: Tue, 9 Sep 2025 16:24:53 +0800
Subject: [PATCH] [Neutron] ovn adopt uwsgi

As [1] fixed and [2] introduced, we should use uwsgi for Neutron server
with OVN backend now.

[1] https://bugs.launchpad.net/neutron/+bug/1912359
[2] https://bugs.launchpad.net/neutron/+bug/2087953

Change-Id: I81ef7571afafaf6dedcdc2df534bd44d891c4803
Signed-off-by: ricolin <rlin@vexxhost.com>
---
 neutron/templates/bin/_neutron-server.sh.tpl  | 32 +++++++------------
 .../notes/neutron-3c33aea435f7ab8a.yaml       | 14 ++++++++
 2 files changed, 25 insertions(+), 21 deletions(-)
 create mode 100644 releasenotes/notes/neutron-3c33aea435f7ab8a.yaml

diff --git a/neutron/templates/bin/_neutron-server.sh.tpl b/neutron/templates/bin/_neutron-server.sh.tpl
index be4b254a..157a3dc4 100644
--- a/neutron/templates/bin/_neutron-server.sh.tpl
+++ b/neutron/templates/bin/_neutron-server.sh.tpl
@@ -18,36 +18,26 @@ set -ex
 COMMAND="${@:-start}"

 function start () {
-# (ricolin): Currently ovn have issue with uWSGI,
-# let's keep using non-uWSGI way until this bug fixed:
-# https://bugs.launchpad.net/neutron/+bug/1912359
+  confs="--config-file /etc/neutron/neutron.conf"
 {{- if ( has "ovn" .Values.network.backend ) }}
-  start_ovn
-{{- else }}
-  exec uwsgi --ini /etc/neutron/neutron-api-uwsgi.ini
-{{- end }}
-}
-
-function start_ovn () {
-  exec neutron-server \
-        --config-file /etc/neutron/neutron.conf \
-{{- if ( has "ovn" .Values.network.backend ) }}
-        --config-file /tmp/pod-shared/ovn.ini \
+  confs+=" --config-file /tmp/pod-shared/ovn.ini"
 {{- end }}
 {{- if .Values.conf.plugins.taas.taas.enabled }}
-        --config-file /etc/neutron/taas_plugin.ini \
+  confs+=" --config-file /etc/neutron/taas_plugin.ini"
 {{- end }}
 {{- if ( has "sriov" .Values.network.backend ) }}
-        --config-file /etc/neutron/plugins/ml2/sriov_agent.ini \
+  confs+=" --config-file /etc/neutron/plugins/ml2/sriov_agent.ini"
 {{- end }}
 {{- if .Values.conf.plugins.l2gateway }}
-        --config-file /etc/neutron/l2gw_plugin.ini \
+  confs+=" --config-file /etc/neutron/l2gw_plugin.ini"
 {{- end }}
 {{- if ( has "tungstenfabric" .Values.network.backend ) }}
-        --config-file /etc/neutron/plugins/tungstenfabric/tf_plugin.ini
+  confs+=" --config-file /etc/neutron/plugins/tungstenfabric/tf_plugin.ini"
 {{- else }}
-        --config-file /etc/neutron/plugins/ml2/ml2_conf.ini
+  confs+=" --config-file /etc/neutron/plugins/ml2/ml2_conf.ini"
 {{- end }}
+
+  exec uwsgi --ini /etc/neutron/neutron-api-uwsgi.ini --pyargv " $confs "
 }

 function stop () {
